app/views/tags/show.mobile.haml

```diff
@@ -5,13 +5,14 @@
 %h1
   = @stream.display_tag_name
 - if user_signed_in?
-  - unless tag_followed?
-    .btn.btn-success.tag_following_action{data: {name: @stream.tag_name}}
-      = t(".follow", tag: @stream.tag_name)
+  -# rubocop:disable Style/IdenticalConditionalBranches
+  - if tag_followed?
+    .btn.btn-danger.tag_following_action{data: {name: @stream.tag_names}}
+      = t(".stop_following", tag: @stream.tag_names)
   - else
-    .btn.btn-danger.tag_following_action{data: {name: @stream.tag_name}}
-      = t(".stop_following", tag: @stream.tag_name)
-
+    .btn.btn-success.tag_following_action{data: {name: @stream.tag_names}}
+      = t(".follow", tag: @stream.tag_names)
```

Comment:
This code is for mobile, but the problem I describe exist for both mobile and desktop implementations.

So when I search for "#photo #cats", what I see there on the button to follow and stop following the tags. If I interact with the button it subscribes me to the combined tag of `#photocats` which is probably not what the user would expect to happen

![image](https://user-images.githubusercontent.com/10187586/178120969-f6838d16-7270-4d75-849c-4d34366fd6a6.png)
![image](https://user-images.githubusercontent.com/10187586/178121017-5dba216f-089e-4ffa-a7c1-8b7f9097603f.png)